icu4j/main/tests/core/src/com/ibm/icu/dev/test/format/TestMessageFormat.java

```diff
@@ -2122,4 +2124,49 @@ public void TestMessageFormatNumberSkeleton() {
             assertEquals(messagePattern, expected, sb.toString());
         }
     }
+
+    private static void doTheRealDateTimeSkeletonTesting(Date date, Object[][] cases) {
```

Comment:
This works, but I think it's still worth to simplify this function as I suggested for C++ (take one triple of values at a time instead of an array, call this function multiple times) for readability & simplicity -- and to avoid the type casts.

Reply:
Done

Reply:
actually not done -- why not?

Reply:
Apologies, did it last night, at home, but forgot to push.
And I said "Done" based on memory, without checking.